runtime/bigarray.c

```diff
@@ -154,7 +155,7 @@ CAMLexport void caml_ba_finalize(value v)
     if (b->proxy == NULL) {
       free(b->data);
     } else {
-      if (-- b->proxy->refcount == 0) {
+      if (caml_atomic_refcount_decr(&b->proxy->refcount) == 0) {
```

Comment:
I thought `refcount_decr` returns the value before decrement.  In this case it should be `== 1`, not `== 0`.

Reply:
Indeed, that was a stupid mistake. Fixed.